src/site/topics/bags/bag-implementations.rst

```diff
@@ -2,419 +2,299 @@
+    :download:`ArrayIndexedBag </../main/java/ArraySortedBag.java>` class for the full implementation.
 
-    public class ArraySortedBag<T extends Comparable<? super T>> implements SortedBag<T> {
 
-        private static final int DEFAULT_CAPACITY = 100;
-        private static final int NOT_FOUND = -1;
-        private T[] bag;
-        private int rear;
+.. literalinclude:: /../main/java/ArraySortedBag.java
+    :language: java
+    :lineno-match:
+    :lines: 1-25
+    :emphasize-lines: 12
 
-        public ArraySortedBag() {
-            this(DEFAULT_CAPACITY);
-        }
 
-        @SuppressWarnings("unchecked")
-        public ArraySortedBag(int initialCapacity) {
-            bag = (T[]) new Comparable[initialCapacity];
-            rear = 0;
-        }
+* Notice ``<T extends Comparable<? super T>>``
+* There is a little bit to unpack here
 
-* This looks nearly the same as the ``ArrayIndexedBag`` implementation, but we see one major difference
+* First, when something extends `Comparable <https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Comparable.html>`_, it means that the type has some defined ordering
 
-    ``<T extends Comparable<? super T>>``
+    * The method ``compareTo`` is implemented for the type
 
-* There is a little bit to unpack here
 
-* First, when something extends `Comparable <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Comparable.html>`_, it means that the type has some defined ordering
-    * The method ``compareTo`` is implemented for the type
+* If ``x.compareTo(y)`` is called
 
-* If we call ``x.compareTo(y)``
     * Return a negative integer if ``x < y``
     * Return zero if ``x == y``
     * Return a positive integer if ``x > y``
 
-* When we have something that is extending ``Comparable<T>``, that means we can compare ``this`` to some type ``T``
+
+* When something extends ``Comparable<T>``, that means ``this`` can be compared to some type ``T``
+
     * ``this`` can be compared to things of type ``T``, but not the other way around
 
-* By saying ``T extends Comparable<T>``, this means that the type ``T`` we want in our ``ArraySortedBag`` can be compared to things of type ``T`` to provide some defined ordering
-    * Which is needed, if we want to sort things
 
-* Finally, with ``Comparable<? super T>``, we are saying that ``T`` or one of its superclasses must extend ``Comparable``
+* ``T extends Comparable<T>`` means that the type ``T`` can be compared to things of type ``T`` to provide some defined ordering
+
+    * Which is needed, if the elements are to be sorted
+
+
+* Finally, ``Comparable<? super T>`` means that ``T``, or one of its superclasses, must extend ``Comparable``
+
     * ``?`` is a *wildcard*
-    * If our type ``T`` does not extend ``Comparable`` and have a ``compareTo`` method written, that may be fine as long as one of its superclasses does
+    * It is fine if ``T`` does not extend ``Comparable`` as long as one of its superclasses does
```

Comment:
That is not accurate. `T extends Comparable` already means `T` or one of it's superclasses inherit from `Comparable`. The type argument of `Comparable` talks about what type this thing is comparable to. `T extends Comparable<U>` means `int T#compareTo(U)` exists. So, `<? super T>` means _comparable to `T` or a superclass of `T`_ (if superclass, then it means it is also comparable to `T` through inheritance).

So, putting it all together. This signature allows types like:
```
class Number extends Comparable<Number>
class Int extends Comparable<Number>
```
but does not allow:
```
class Int extends Comparable<String>
```